src/RegexProvider/RegexProvider.fs

```diff
@@ -11,10 +11,20 @@ module internal TypedRegex =
     let typedRegex() = 
         let regexType = erasedType<Regex> thisAssembly rootNamespace "Regex"
         regexType.DefineStaticParameters(
-            parameters = [ProvidedStaticParameter("pattern", typeof<string>)], 
+            parameters =
+                [
+                    ProvidedStaticParameter("pattern", typeof<string>)
+                    ProvidedStaticParameter("generatedMethodsPrefix", typeof<string>, "")
```

Comment:
I don't quite like that user can provide any prefix. He will have to provide the same string again and again, for each type. Maybe it'd be better to add `Typed` prefix by default and no prefix if a boolean static parameter provided, like this:
```fsharp
// "TypedXXX"
type WordRegex = Regex< @"(?<Word>\w+)">
// The old behavior, for backward compatibility
type WordRegex = Regex< @"(?<Word>\w+)", NoMethodPrefix = true>
```